### PR TITLE
Reorganize TransportLayer, CommUtil, BlockStore helpers

### LINKED ISSUE #2970: Reorganize TransportLayer, CommUtil, BlockStore helpers
## Overview
Idea is to have only basic operations on traits. Operations that can be defined by the base should be put in extension methods (syntax).
CommUtil can be refactored in this way and extensions can be defined on TransportLayer instance (+ config).

### PR DESCRIPTION
**Pull request by tgrospic**
## Overview
This PR is not changing any functionality it's just refactoring of CommUtil interface to reduce basic functions which are moved to TransportLayer extensions (syntax).

Also two methods to get block data unsafe (with applicative error) are transformed to extension methods.

Implicit classes for extension methods for KeyValueStore now extends AnyVal for slightly better optimization. 

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/EstimatorHelper.scala [hunk 1]
```diff
@@ -115,14 +116,13 @@ object EstimatorHelper {
...
       blockAncestorsMeta: List[BlockMetadata]
   ): F[BlockEvents] =
     for {
-      maybeAncestors <- blockAncestorsMeta.traverse(
-                         blockAncestorMeta => BlockStore[F].get(blockAncestorMeta.blockHash)
-                       )
-      ancestors = maybeAncestors.flatten
+      ancestors <- blockAncestorsMeta.traverse(
+                    blockAncestorMeta => BlockStore[F].getUnsafe(blockAncestorMeta.blockHash)
+                  )
```
Comment (tgrospic) [inline casper/src/main/scala/coop/rchain/casper/EstimatorHelper.scala:125]: This code can not throw errors which should be what we want and not to skip ancestors if not in the store.

Comment (nzpr) [reply casper/src/main/scala/coop/rchain/casper/EstimatorHelper.scala:125]: Why here you changed Monad to Sync?

Comment (tgrospic) [reply casper/src/main/scala/coop/rchain/casper/EstimatorHelper.scala:125]: Because of use of `Sync.raiseError` inside `BlockStore[F].getUnsafe`.

**Review by nzpr — APPROVED** [review]
LGTM
I rebased https://github.com/rchain/rchain/pull/2943 on this PR

**Review by zsluedem — APPROVED** [review]
This looks great to me. I got better understanding on syntax.
So we should define trait with the most generic method and then use syntax to combine these methods. Does that make sense?

**Comment by tgrospic** [comment]
> This looks great to me. I got better understanding on syntax.
> So we should define trait with the most generic method and then use syntax to combine these methods. Does that make sense?

@zsluedem Yes. Generic methods on a trait can be seen as a definition of the core language (rules) and syntax (extensions) are like sentences written in that language.

**Comment by tgrospic** [comment]
bors merge

**Comment by nzpr** [comment]
bors merge

**Comment by tgrospic** [comment]
bors retry

**Comment by tgrospic** [comment]
bors retry

**Pull request merged**